src/test/java/seedu/address/model/AddressBookTest.java

```diff
@@ -49,7 +50,18 @@ public void resetData_withDuplicatePersons_throwsDuplicatePersonException() {
         Person editedAlice = new PersonBuilder(ALICE).withAddress(VALID_ADDRESS_BOB).withTags(VALID_TAG_HUSBAND)
                 .build();
         List<Person> newPersons = Arrays.asList(ALICE, editedAlice);
-        AddressBookStub newData = new AddressBookStub(newPersons);
+        AddressBookStub newData = new AddressBookStub(newPersons, null);
+
+        assertThrows(DuplicatePersonException.class, () -> addressBook.resetData(newData));
+    }
+
+    @Test
+    public void resetData_withDuplicateInternships_throwsDuplicateInternshipException() {
+        // Two Internships with the same identity fields
+        Person editedAlice = new PersonBuilder(ALICE).withAddress(VALID_ADDRESS_BOB).withTags(VALID_TAG_HUSBAND)
+                .build();
+        List<Person> newPersons = Arrays.asList(ALICE, editedAlice);
+        AddressBookStub newData = new AddressBookStub(newPersons, null);
 
```

Comment:
I understand that you are still using the Person class because we have not fully transitioned away from AB3. Remember to make the necessary amendments next time.